client/oidc/src/test/java/io/quarkiverse/openapi/generator/oidc/ReactiveOAuth2AuthenticationProviderTest.java

```diff
@@ -0,0 +1,124 @@
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import io.quarkiverse.openapi.generator.AuthConfig;
+import io.quarkiverse.openapi.generator.oidc.providers.OAuth2AuthenticationProvider;
+import io.quarkus.oidc.client.Tokens;
+import io.smallrye.mutiny.Uni;
+
+@ExtendWith(MockitoExtension.class)
+public class ReactiveOAuth2AuthenticationProviderTest {
+    private static final String OPEN_API_FILE_SPEC_ID = "open_api_file_spec_id_json";
+    private static final String AUTH_SCHEME_NAME = "auth_scheme_name";
+
+    private static final String ACCESS_TOKEN = "REACTIVE_ACCESS_TOKEN";
+
+    private static final String PROPAGATED_TOKEN = "PROPAGATED_TOKEN";
+
+    private static final String HEADER_NAME = "HEADER_NAME";
+    @Mock
+    private ResteasyReactiveClientRequestContext reactiveRequestContext;
+    private MultivaluedMap<String, Object> headers;
+
+    private ReactiveOidcClientRequestFilterDelegate reactiveDelegate;
+    private static final Tokens token = new Tokens(ACCESS_TOKEN, Long.MAX_VALUE, null, "", Long.MAX_VALUE, null, "");
+    private static final Uni<Tokens> uniToken = Uni.createFrom().item(token);
+
+    private OAuth2AuthenticationProvider provider;
+
+    @BeforeEach
+    void setUp() {
+        headers = new MultivaluedHashMap<>();
+        Mockito.lenient().doReturn(headers).when(reactiveRequestContext).getHeaders();
+
+        reactiveDelegate = Mockito.mock(ReactiveOidcClientRequestFilterDelegate.class);
+        try {
+            Mockito.lenient().doCallRealMethod().when(reactiveDelegate).filter(Mockito.any(ClientRequestContext.class));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        Mockito.lenient().doCallRealMethod().when(reactiveDelegate).filter(reactiveRequestContext);
+        Mockito.lenient().when(reactiveDelegate.getTokens()).thenReturn(uniToken);
+
+        provider = createReactiveProvider();
+    }
+
+    protected OAuth2AuthenticationProvider createReactiveProvider() {
+        return new OAuth2AuthenticationProvider(AUTH_SCHEME_NAME, OPEN_API_FILE_SPEC_ID, reactiveDelegate, List.of());
+    }
+
+    protected void assertHeader(MultivaluedMap<String, Object> headers, String headerName, String value) {
+        Assertions.assertThat(headers.getFirst(headerName))
+                .isNotNull()
+                .isEqualTo(value);
+    }
+
+    static Stream<Arguments> filterWithPropagationTestValues() {
+        return Stream.of(
+                Arguments.of(null, "Bearer " + PROPAGATED_TOKEN),
+                Arguments.of(HEADER_NAME, "Bearer " + PROPAGATED_TOKEN));
+    }
+
+    @Test
+    void filterReactive() throws IOException {
+        filter(provider, "Bearer " + ACCESS_TOKEN);
+    }
+
+    private void filter(OAuth2AuthenticationProvider provider, String expectedAuthorizationHeader) throws IOException {
+        provider.filter(reactiveRequestContext);
+        assertHeader(headers, HttpHeaders.AUTHORIZATION, expectedAuthorizationHeader);
```

Comment:
`filter` is a non-blocking method and then, immediately, you are checking the headers in the `assertHeader`.
Potentially header couldn't be there yet, making the test to be flaky. Of course it's needed by Quarkus but it's not testing-friendly.

In this case, you could have a `filterAsync` method in the `ReactiveOidcClientRequestFilterDelegate` provider, returning a `Uni<Void>`
And then use here 
`provider.filterAsync(reactiveRequestContext).await().indefinitely();` to make more reliable this test.

Wdyt?

Reply:
not sure how I could use a `CountDownLatch` as I do not want to add code just for testing purposes in the `filter` part handling the response
Plus, the `getTokens` method is mocked and should return instantly.

But I do agree, in rare case, there may be sync issue and the test may fail; to avoid that, I mocked the `RestClientRequestContext` that executes the `suspend` and `resume` methods which are used in the  `filter` . THen I added a simple `while-sleep` loop to wait until the request is resumed. The loop may not be the most efficient solution, but for this case I believe it's enough as the impact will be very low, WDYT?